packages/wonder-blocks-form/src/components/labeled-text-field.js

```diff
@@ -71,36 +102,64 @@ export default class LabeledTextField extends React.Component<Props, State> {
     constructor(props: Props) {
         super(props);
         this.state = {
-            value: props.initialValue ? props.initialValue : "",
+            value: props.initialValue || "",
             error: null,
             focused: false,
         };
     }
 
-    handleOnFocus: (
-        event: SyntheticFocusEvent<HTMLInputElement>,
-    ) => mixed = () => {
+    handleValidation: (errorMessage: ?string) => mixed = (errorMessage) => {
+        const {onValidation} = this.props;
+        this.setState({error: errorMessage});
+        if (onValidation) {
+            onValidation(errorMessage);
+        }
+    };
+
+    handleChange: (newValue: string) => mixed = (newValue) => {
+        const {onChange} = this.props;
         this.setState({
-            focused: true,
+            value: newValue,
         });
+        if (onChange) {
+            onChange(newValue);
+        }
```

Comment:
Should this be invoked once the state has been applied (as in using the `setState` callback)?